service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcService.java

```diff
@@ -0,0 +1,35 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.storage.StorageConnectorService;
+import uk.nhs.adaptors.gp2gp.ehr.request.EhrExtractRequestHandler;
+
+@Service
+@Slf4j
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GpcService {
```

Comment:
Could this class be inlined into the task executor? All the task executor does currently is call this class' sole method. Might be an unnecessary level in indirection.